### DISCUSSION TITLE
Rate of Change (ROC) and Momentum Oscillator

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#242] [Indicators]
Community discussion about [Rate of Change (ROC), Momentum Oscillator, and ROC with Bands](https://dotnet.stockindicators.dev/indicators/Roc/#content)

**Comment by kavatari** [comment]
Hi @DaveSkender
currently, I am trying to understand some deviations in the Momentum (14) calculation. Using the same data set for the Momentum (14) calculation, I have following deviations between this .NET library and Highcharts...
For the calculation I provide 50 quotes to the library...
**The results:** 
**.NET lib:** -2.72
**Highcharts:** -3.38

Could you give me a hint why is there a deviation?

**Reply by DaveSkender** [reply]
This one is easy to test since [momentum](https://en.wikipedia.org/wiki/Momentum_(technical_analysis)) is simply the change in price over your lookback period (14 bars).  The only reasonable alternate calculation is whether or not it's measuring _raw price change_ or _percent price change_.  Our [Rate of Change (ROC) indicator](https://dotnet.stockindicators.dev/indicators/Roc/#content) reflects the percent price change.

**Reply by DaveSkender** [reply]
I’m looking at this closer and I do see some other variations out there where the Momentum Oscillator is defined slightly differently from ROC.  Momentum alone should still be just a raw price change.

**Reply by DaveSkender** [reply]
This is what I’m seeing (rare, alternative definition for Momentum Oscillator):

`Momentum Oscillator = (Price today / Price n periods ago) x 100`

whereas most places define the Momentum Oscillator the same as our library:

`Rate of Change = ((Price today / Price n periods ago) - 1) x 100`

These are effectively the same but scaled differently.

And for clarity

`Momentum = Price today - Price n periods ago`

**Reply by kavatari** [reply]
Thanks for the clarification, I will dig into it and do some tests :) We keep you posted here.

**Reply by kavatari** [reply]
I did the math and looked once again at Highcharts...they have also the "Rate of Change" as a separate indicator and the results are the same compared with this wonderful .NET lib.
@DaveSkender thanks once again for the clarification.

**Reply by DaveSkender** [reply]
Were you able to determine what their `Momentum` indictor does?  I’m curious, but not enough to go look myself.  😬

**Reply by kavatari** [reply]
Yes, it does exactly this: 
`Momentum = Price today - Price n periods ago`

So, it was just a confusion on my side and I just needed a clarification.
